ladybug/compass.py

```diff
@@ -0,0 +1,259 @@
+                  'WNW', 'NW', 'NNW')
+    ALTITUDES = (10, 20, 30, 40, 50, 60, 70, 80)
+    PI = math.pi
+
+    def __init__(self, radius=100, center=Point2D(), north_angle=0,
+                 spacing_factor=0.15):
+        """Initialize Compass."""
+        self.radius = radius
+        self.center = center
+        self.north_angle = north_angle
+        self.spacing_factor = spacing_factor
+
+    @property
+    def radius(self):
+        """Get or set a positive number for the radius of the compass."""
+        return self._radius
+
+    @radius.setter
+    def radius(self, value):
+        self._radius = float(value)
+        assert self._radius > 0, \
+            'Compass radius must be a positive number. Got {}.'.format(value)
+
+    @property
+    def center(self):
+        """Get or set a Point2D for the center of the compass in the scene."""
+        return self._center
+
+    @center.setter
+    def center(self, value):
+        assert isinstance(value, Point2D), 'Expected ladybug_geometry Point2D ' \
+            'for Compass center. Got {}.'.format(type(value))
+        self._center = value
+
+    @property
+    def north_angle(self):
+        """Get or set a number between -360 and 360 for the north_angle in degrees."""
+        return math.degrees(self._north_angle)
+
+    @north_angle.setter
+    def north_angle(self, value):
+        self._north_angle = math.radians(float(value))
+        assert -self.PI * 2 <= self._north_angle <= self.PI * 2, \
+            'north_angle value should be between -360 and 360. Got {}.'.format(value)
+
+    @property
+    def spacing_factor(self):
+        """Get or set a number for the fraction of radius occupied by labels and ticks."""
+        return self._spacing_factor
+
+    @spacing_factor.setter
+    def spacing_factor(self, value):
+        self._spacing_factor = float(value)
+        assert self._spacing_factor > 0, \
+            'Compass spacing_factor must be a positive number. Got {}.'.format(value)
+
+    @property
+    def min_point(self):
+        """Get a Point2D for the minimum around the entire compass."""
+        fac = (1 + self.spacing_factor) * self.radius
+        return Point2D(self.center.x - fac, self.center.y - fac)
+
+    @property
+    def max_point(self):
+        """Get a Point2D for the minimum around the entire compass."""
+        fac = (1 + self.spacing_factor) * self.radius
+        return Point2D(self.center.x + fac, self.center.y + fac)
+
+    @property
+    def inner_boundary_circle(self):
+        """Get a Arc2D for the inner circle of the compass.
+        
+        This is essentially a circle with the compass radius.
+        """
+        return Arc2D(self.center, self.radius) 
+
+    @property
+    def all_boundary_circles(self):
+        """Get an array of 3 Arc2Ds for the circles of the compass."""
+        arc2 = Arc2D(self.center, self.radius * (1 + self.spacing_factor * 0.1))
```

Comment:
I'm ok if you want to open an issue about this on ladybug-geometry but, if you do so, it would be good to know what things you see distinguishing these classes from one another since, if we were to implement it now, it would probably just result in an empty class (either circle or arc) that inherits everything from the other class. It would also be good to know how you would recommend dealing with arcs objects that are circles (eg. should we not allow an arc to be a closed and thus get rid of the [is_circle](https://www.ladybug.tools/ladybug-geometry/docs/ladybug_geometry.geometry2d.arc.html#ladybug_geometry.geometry2d.arc.Arc2D.is_circle) property).

Reply:
I assume circle is a subclass of Arc where the angle is always 360 and is_circle always returns true. As simple as that.

Reply:
Ok. So it sounds like you just overwrite `__init__` for the circle to not expose the angular cutoffs, I guess that we can also add properties like `area` that don't really have meaning for an arc. Ok, feel free to open an issue about it on ladybug-geometry and I'll get to it when I can.